Source/UZKFileInfo.m

```diff
@@ -43,6 +43,9 @@ - (instancetype)initWithFileInfo:(unz_file_info64 *)fileInfo filename:(NSString
         
         _compressionMethod = [self readCompressionMethod:fileInfo->compression_method
                                                     flag:fileInfo->flag];
+        
+        uLong permissions = (fileInfo->external_fa >> 16) & 0777U;
```

Comment:
I think you should avoid using such high permissions as 0777. It is recommended to use a maximum of 0755 permissions to prevent malicious code from being corrupted in the zip file.

Reply:
You're misunderstanding what this line does. It's applying a bit mask to the `external_fa` field, basically saying get rid of all bits from this field except the permission bits. I use `777` here because that turns on all of the permission bits, and leaves everything else off.